### PR TITLE
Upgrade Scala sttp client generator to sttp3

### PR DESCRIPTION
**Pull request by borissmidt**
The sttp generator is out of date by atleast a year, sttp2 has been deprecated and i have updated the generator to now support sttp3.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
- [  ] 
@chameleon82

### CODE REVIEW

**Comment by borissmidt** [comment]
https://github.com/softwaremill/tapir/blob/master/examples/src/main/scala/sttp/tapir/examples/BooksExample.scala
I think it might be even better if we added tapir support instead then it will be really close to 'the universal' scala openApi generator. 
I.e. it supports a lot of servers and a lot of clients. So it is like sttp but then reusable for servers.

**Comment by wing328** [comment]
cc @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

**Comment by wing328** [comment]
FYI. I've changed the target from 5.4.x to master as the master is the upcoming 5.4.0 release.

File: samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ class PetApi(baseUrl: String) {
    * @param pet Pet object that needs to be added to the store
    */
   def addPet(pet: Pet
-): Request[Either[ResponseError[Exception], Pet], Nothing] =
+): Request[Either[ResponseException[String, Exception], Pet], Nothing] =
```
Comment (wing328) [inline samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala:36]: @borissmidt thanks for the PR. Is this a breaking change (from `ResponseError` to `ResponseException`) ?

In other words, users need to update their code accordingly after this change, right?

Comment (borissmidt) [reply samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala:36]: I will add a new parameter to check which sttp version is selected and branch depending on that to keep backward compatiblity.

Comment (wing328) [reply samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala:36]: Maybe simply target this change to 6.0.x branch as we'll release 6.0.0 in 2022 Q1

Comment (borissmidt) [reply samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala:36]: Oke good idea!

Comment (borissmidt) [reply samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala:36]: I rebased it on the 6.0.x branch

**Comment by wing328** [comment]
cc @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

**Comment by borissmidt** [comment]
@wing328 Thank you!

**Pull request merged**